src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+Alternate Code Example
+=======================
+
+Let’s move beyond our "Hello, World" example and explore a simple temperature
+conversion program. We want our function to convert a Fahrenheit temperature to
+Celsius.
+
+Temperature Conversion
+-----------------------
+
+If you need to review the steps for creating a new IntelliJ project, refer
+back to the :ref:`First Java Program <create-new-java-project>` instructions.
+
+#. Create a new project and call it ``TempConverter``.
+#. Right-click (or control-click) on the ``src`` folder. Select *New*, then
+   *Java Class*.
+
+   .. figure:: figures/create-new-class.png
+      :alt: Create the TempConverter class.
+
+#. You should see the following empty class in the editor panel:
+
+   .. figure:: figures/TempConverter-class.png
+      :alt: New TempConverter class.
+
+#. Cut and paste the following code into the editor. We will analyze the
+   different statements in a moment.
+
+   .. sourcecode:: java
+      :linenos:
+
+      import java.util.Scanner;
+
+      public class TempConverter {
+         public static void main(String[] args) {
+            double fahrenheit;
+            double celsius;
+            Scanner input;
+
+            input = new Scanner(System.in);
+            System.out.println("Enter the temperature in Fahrenheit: ");
+            fahrenheit = input.nextDouble();
+            input.close();
+
+            celsius = (fahrenheit - 32) * 5/9;
+            System.out.println("The temperature in Celsius is: " + celsius + "°C");
+         }
+      }
+
+#. Build and run the program to verify that it works. Entering a Fahrenheit
+   temperature of ``212`` yields the result, ``The temperature in Celsius is:
+   100.0°C``.
+
+There are several new concepts introduced in this example. We will look
+at them in the following order:
+
+#. The ``import`` statement
+#. Declaring variables
+#. Collecting input with the Scanner class
+
+``import``
+-----------
+
+The ``import`` statement in Java allows us to access the class, methods, and
+data stored a different file. ``import`` tells the compiler that we are going
+to use a shortened version of the class name. In this example, we are going to
+use the class ``java.util.Scanner``, but we can refer to it as just
+``Scanner``. We could use the ``java.util.Scanner`` class without any problem
+and without any import statement, provided that we always referred to it by its
+full name.
+
+In Java, you can use any class that is available WITHOUT having to import the
+class, but you must adhere to two very important conditions:
+
+#. The ``javac`` and ``java`` commands must know that the class exists.
```

Comment:
this line is confusing to me. i think the point is to say that the class needs to exist but i'd suggest avoiding stating that the compiler and java commands know this

Reply:
I made an attempt to clarify this.  Let me know what you think.